packages/@ember/-internals/glimmer/lib/helpers/fn.ts

```diff
@@ -6,6 +6,77 @@ import { InternalHelperReference } from '../utils/references';
 import buildUntouchableThis from '../utils/untouchable-this';
 
 const context = buildUntouchableThis('`fn` helper');
+
+/**
+@module ember
+*/
+
+/**
+  The `fn` helper allows you to ensure a function that you are passing off
+  to another component, helper, or modifier has access to arguments that are
+  available in the template.
+
+  For example, if you have an `each` loop over a number of items and need to
+  pass a function to a component invoked within the loop but that function needs
+  access to the item.
+
+  ```app/components/items-list.js
+  import Component from '@glimmer/component';
+  import { action } from '@ember/object';
+
+  export default class ItemsList extends Component {
+    @action
+    handleSelected(item) {
+      // ...snip...
+    }
+  }
+  ```
```

Comment:
I think it's helpful (but not necessary) to show the template first, and then the component js.